### PR TITLE
fix: no redundant space in header `Location` for `HTTPRoute` with requestRedirect filter

### PR DESCRIPTION
**Pull request by programmer04**
**What this PR does / why we need it**:

As described in the reported issue https://github.com/Kong/kubernetes-ingress-controller/issues/6851 header (port `:443` is missing due to changes from https://github.com/Kong/kubernetes-ingress-controller/pull/6804)

```
Location:  https://demo.example.com/path
```

two spaces after colon instead of expected

```
Location: https://demo.example.com/path
```

Unfortunately, not all programs ignore whitespaces in headers (they should), thus it's problematic.

This PR fixes it. 

**Which issue this PR fixes**:

Fixes https://github.com/Kong/kubernetes-ingress-controller/issues/6851

**Special notes for your reviewer**:

The actual root cause was [here](https://github.com/Kong/kubernetes-ingress-controller/pull/6855/files#diff-df2a67c9ede8d32433f70281fa4979f7c97dc5229fbf51458d01be729ee2bedeL1132-L1134), all other changes are a parto of an opprotunistic refactor to avoid such problems in the future.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Comment by randmonkey** [comment]
Please check if there is still the whitespace in #6804.

File: CHANGELOG.md [hunk 1]
```diff
@@ -189,6 +189,10 @@ Adding a new version? You'll need three changes:
 - Fixed validation of JWT credentials using non HMAC algorithms where `secret`
   field was incorrectly required.
   [#6848](https://github.com/Kong/kubernetes-ingress-controller/pull/6848)
+- There is no redundant space in header `Location` when `HTTPRoute` with
+  requestRedirect filter is used.
+  [#6855](https://github.com/Kong/kubernetes-ingress-controller/pull/6855)
+
```
Comment (czeslavo) [inline CHANGELOG.md:195]: As we're on the redundant whitespace topic 😄 
```suggestion
  [#6855](https://github.com/Kong/kubernetes-ingress-controller/pull/6855)
```

**Review by randmonkey — CHANGES_REQUESTED** [review]
Hold here to prevent conflict with #6804.

**Review by czeslavo — COMMENTED** [review]
👍

**Review by czeslavo — DISMISSED** [review]

**Comment by programmer04** [comment]
Your PR has been merged @randmonkey, I've rebased this and it's ready to merge too

**Review by czeslavo — APPROVED** [review]

**Review by randmonkey — APPROVED** [review]

**Pull request merged**